src/System.Net.NameResolution/src/System/Net/Dns.cs

```diff
@@ -59,6 +61,7 @@ public static IPHostEntry GetHostEntry(string hostNameOrAddress)
             {
                 if (address.Equals(IPAddress.Any) || address.Equals(IPAddress.IPv6Any))
                 {
+                    if (NetEventSource.IsEnabled) NetEventSource.Error(hostNameOrAddress, $"Invalid address {address}");
```

Comment:
I'm not understanding why we're passing these objects in as the first argument to these events.  That's generally reserved for the instance with which the message is associated.  We want to treat these strings in that capacity?  That's probably fine, but it's not what we've done in the past.

Reply:
The goal was ability to match enter and exit entries as well as error messages. 
The reason why this is passed in as that it will show in thisOrContextObject. 
So yes, I use it is instance in lack of better anchor. 
Is there better way how to link them together? I think best way how to see the impact is to look at the generated message. I use Microsoft.Diagnostics.Tracing to generate output.

Reply:
note that for synchronous calls you can use thread id but that does not work for Aync.